### ISSUE TITLE
flag warning for aria-hidden on element that has hidden attribute

### ISSUE THREAD

**Issue by carmacleod** [IBMa/equal-access#350]
Steps:
- go to https://w3c.github.io/html-aria/tests/aria-hidden-valid-element-test-hidden-attr.html
- run accessibility checker
- there should be a warning flagged on all elements that have both aria-hidden and hidden

**Comment by scottaohara** [comment]
here's an updated test case:
https://w3c.github.io/html-aria/tests/aria-hidden-and-hidden.html

the difference here is that the new test case also includes all focusable elements, like inputs, buttons, links, etc.

**Comment by MHoov** [comment]
As of June 1
- [x] general rule description (SME & Dev)
- [x] testcases written (Dev)
- [ ] rule developed  (Dev)
- [ ] message written (SME)
- [ ] help written (SME)

**Comment by tombrunet** [comment]
I don't agree with this one. Anything marked as `display:none`, `visibility:hidden`, or `hidden` is not available visibly or to an AT regardless of what any other attribute says. This behavior is well-defined and isn't related to aria-hidden in any way since it affects visibility in addition to inclusion in the accessibility hierarchy.

You might do this intentionally if you're doing a show/hide on something, but don't want it visible to an AT user.

Can you better justify this?

**Comment by tombrunet** [comment]
As an addition - a lot of the checked vs aria-checked conflict checking makes sense because those attributes map to the same fields. It's a conflict / redundancy question. However, despite the naming being similar, `hidden` and `aria-hidden` are two different things. `aria-hidden` isn't an ARIA version of the native `hidden` attribute.

**Comment by scottaohara** [comment]
There are situations where developers have used both the hidden attribute and the aria-hidden=true attribute and toggled the hidden's display with CSS, exposing the element - but it is not then exposed to AT.  I understand your point that might be done on purpose, but in my experiences it has not been.

The recommendation here is to flag a warning, not a failure.  I don't see why that's objectionable?   It's either redundant to what the hidden attribute already does, or it could be problematic if a developer alters one attribute state but not the other.

**Comment by aliunwala** [comment]
Triage meeting comments:
We might want to do this to give the user's the feedback and warn them. But we also might want to have a way to turn these warnings off before we do, as to not clutter the screen with extra content.

**Comment by marcjohlic** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Equal-Access-61ae71f9f8b4ec001e136462/issues/IBMa/equal-access/350?planning-poker) with ZenHub @aliunwala @drjoho @mbgower @philljenkins @shunguoy @tombrunet

**Comment by shunguoy** [comment]
elements with a hidden attribute are not triggered from the context.

**Comment by tombrunet** [comment]
@scottaohara So what would you do if you toggled the display of a decorative svg? You would have `<svg aria-hidden="true">` because it's decorative. Now you add hidden to hide the display and get `<svg hidden aria-hidden="true">`. This isn't redundant. This really isn't in the same category as an HTML attribute that maps to the same ARIA property.

Also, we generally don't report issues in non-displayed content because it's not in the tree. This would add a ton of weird special casing in order to report a recommendation for something that may not even be an error. Going to close this as a won't change.

**Comment by tombrunet** [comment]
Also note for the testcase above...  when the interactive element is no longer 'hidden', it will flag a different rule because interactive content cannot be aria-hidden.

**Comment by scottaohara** [comment]
hi @tombrunet.  thanks for the update and understood on your point.  

Since filing this issue, changes to `hidden` to no longer be a boolean attribute, but to also allow for this `until-found` value would likely be a more important use case to review.  Based on Chrome's current implementation, it seems that if both are used, the `until-found` feature it negated due to the `aria-hidden=true` usage.  But we will be keeping an eye on how this is implemented in other user agents as well.
